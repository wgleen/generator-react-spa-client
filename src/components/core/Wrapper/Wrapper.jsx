import React from 'react'
import PropTypes from 'prop-types'
import ReduxProvider from '../Redux/ReduxProvider'
import ThemeProvider from '../Theme/ThemeProvider'
import RouterProvider from '../Router/RouterProvider'

const Wrapper = (props) => {
  const { children } = props

  return (
    <>
      <ReduxProvider>
        <ThemeProvider>
          <RouterProvider>{children}</RouterProvider>
        </ThemeProvider>
      </ReduxProvider>
    </>
  )
}

Wrapper.defaultProps = {
  children: null
}

Wrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node])
}

export default Wrapper
