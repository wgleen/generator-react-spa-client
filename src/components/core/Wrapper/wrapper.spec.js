import React from 'react'
import { mount } from 'enzyme'
import Wrapper from './Wrapper'

describe('<Wrapper />', () => {
  it('should render correctly', () => {
    const component = mount(<Wrapper />)

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  it('should render ReduxProvider provider', () => {
    const component = mount(<Wrapper />)

    expect(component.find('ReduxProvider').exists()).toBeTrue()

    component.unmount()
  })

  it('should render ThemeProvider provider', () => {
    const component = mount(<Wrapper />)

    expect(component.find('ThemeProvider').exists()).toBeTrue()

    component.unmount()
  })

  it('should render RouterProvider provider containing the children prop given', () => {
    const children = <>Text test</>

    const component = mount(<Wrapper>{children}</Wrapper>)

    expect(component.find('RouterProvider').text()).toStrictEqual('Text test')

    component.unmount()
  })
})
