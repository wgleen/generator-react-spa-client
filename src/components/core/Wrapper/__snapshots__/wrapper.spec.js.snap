// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render correctly 1`] = `
<Wrapper>
  <ReduxProvider>
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ThemeProvider>
        <Component
          theme={
            Object {
              "colors": Object {
                "blue": "#2e7baf",
                "gray": "#737373",
                "grayDark": "#4d4d4d",
                "grayDarkest": "#0d0d0d",
                "grayLight": "#b3b3b3",
                "grayLightest": "#e5e5e5",
                "green": "#3cab3a",
                "primary": "#ec0000",
                "primaryDark": "#b90000",
                "primaryDarkest": "#860000",
                "primaryLight": "#ff8686",
                "primaryLightest": "#ffd2d2",
                "red": "#ff3225",
                "secondary": "#43bccd",
                "secondaryDark": "#2e9faf",
                "secondaryDarkest": "#247a86",
                "secondaryLight": "#94d9e2",
                "secondaryLightest": "#d0eef2",
                "yellow": "#e6e328",
              },
              "typography": Object {
                "fontFamilyPrimary": "Open Sans",
                "fontWeightPrimary": 300,
                "lineHeight": "normal",
                "textAlign": "left",
              },
            }
          }
        >
          <RouterProvider>
            <BrowserRouter>
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              />
            </BrowserRouter>
          </RouterProvider>
        </Component>
      </ThemeProvider>
    </Provider>
  </ReduxProvider>
</Wrapper>
`;
